src/components/DataCollector/index.js

```diff
@@ -31,10 +31,8 @@ const createDataCollector = () => {
       }
     });
 
-    const responsePromise = Promise.resolve()
-      .then(() => {
-        return lifecycle.onBeforeDataCollection(payload, responsePromise);
-      })
+    return lifecycle
+      .onBeforeDataCollection(payload)
```

Comment:
@jonsnyder This is a change you should be aware of. It would take my quite a while to write about why it was needed and I probably won't communicate it clearly, but let's talk with our mouth holes when you get back. If we need to add it back in, we'll need to do some additional thinking about how to work out some timing things.

Reply:
I put that in there so that components could get the matching response without having to match up requests and responses in onResponse.  We are not using it, so it's okay to remove.